### PR TITLE
[TE] anomaly detection model downloader

### PR DESCRIPTION
**Pull request by jihaozh**
- The model downloader interface. Its implementation is expected to download model files (e.x., trained TensorFlow models), to a local path.
- Model downloader manager class. It initializes the model downloader via reflection and triggers the model downloader to run periodically.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/model/download/ModelDownloaderManager.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package org.apache.pinot.thirdeye.model.download;
```
Comment (akshayrai) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/model/download/ModelDownloaderManager.java]: nit: license header

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/model/download/ModelDownloaderManager.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      } catch (Exception e) {
+        LOG.warn("Failed to initialize model downloader {}", config.getClassName(), e);
+      }
+    }
+  }
+
+  /**
+   * start the model downloader manager
+   */
+  public void start() {
```
Comment (akshayrai) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/model/download/ModelDownloaderManager.java]: Where will this run? every worker?

Comment (jihaozh) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/model/download/ModelDownloaderManager.java]: yes. every worker that needs to run the model

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/model/download/ModelDownloaderManager.java (hunk 1 above)
Comment (jihaozh) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/model/download/ModelDownloaderManager.java]: added

**Pull request merged**